### PR TITLE
Bug 1844471: bump kubernetes to 1.19.0-rc.2

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
it looks like `e2e-aws` is broken with `exec: python2: not found` -  should be fixed by https://bugzilla.redhat.com/show_bug.cgi?id=1873414

File: pkg/cmd/openshift-apiserver/openshiftapiserver/config.go [hunk 1]
```diff
@@ -128,7 +126,7 @@ func NewOpenshiftAPIConfig(config *openshiftcontrolplanev1.OpenShiftAPIServerCon
 		return nil, err
 	}
 
-	authInfoResolverWrapper := webhook.NewDefaultAuthenticationInfoResolverWrapper(nil, genericConfig.EgressSelector, genericConfig.LoopbackClientConfig)
+	// authInfoResolverWrapper := webhook.NewDefaultAuthenticationInfoResolverWrapper(nil, genericConfig.EgressSelector, genericConfig.LoopbackClientConfig)
```
Comment (sttts) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/config.go:129]: remove

File: pkg/cmd/openshift-apiserver/openshiftapiserver/config.go [hunk 2]
```diff
@@ -141,14 +139,16 @@ func NewOpenshiftAPIConfig(config *openshiftcontrolplanev1.OpenShiftAPIServerCon
...
+		/*
+			genericConfig.Config.LoopbackClientConfig,
+			informers.kubernetesInformers,
+			genericapiserveroptions.NewProcessInfo("openshift-apiserver", "openshift-apiserver"),
+			&genericapiserveroptions.WebhookOptions{
+				AuthInfoResolverWrapper: authInfoResolverWrapper,
+				// the openshift-apiserver runs on cluster as a normal pod, accessed by a service, so it should always have access to the service network
+				ServiceResolver: aggregatorapiserver.NewClusterIPServiceResolver(informers.kubernetesInformers.Core().V1().Services().Lister()),
+			},
+		*/
```
Comment (sttts) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/config.go:151]: remove. This is the old, dropped audit webhook code.

File: pkg/build/apiserver/registry/buildconfig/webhook_test.go [hunk 3]
```diff
@@ -18,11 +18,11 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/watch"
-	apirequest "k8s.io/apiserver/pkg/endpoints/request"
 	clientesting "k8s.io/client-go/testing"
-	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	kapi "k8s.io/kubernetes/pkg/apis/core"
 
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
```
Comment (sttts) [inline pkg/build/apiserver/registry/buildconfig/webhook_test.go:24]: nit: import order

File: pkg/image/apis/image/test/conversion_test.go [hunk 4]
```diff
@@ -3,95 +3,29 @@ package v1
 import (
 	"reflect"
 	"testing"
-	"time"
 
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "github.com/openshift/api/image/v1"
```
Comment (sttts) [inline pkg/image/apis/image/test/conversion_test.go:7]: nit: import order

File: pkg/image/apis/image/docker10/types.go [hunk 5]
```diff
@@ -2,54 +2,9 @@ package docker10
 
 import (
 	"time"
-)
-
-// DockerImage is the type representing a container image and its various properties when
-// retrieved from the Docker client API.
-type DockerImage struct {
```
Comment (sttts) [inline pkg/image/apis/image/docker10/types.go:9]: this were duplicated into api repo?

File: pkg/image/apis/image/v1/register.go [hunk 6]
```diff
@@ -4,19 +4,24 @@ import (
...
 
+	"github.com/openshift/api/image/docker10"
+	"github.com/openshift/api/image/dockerpre012"
 	v1 "github.com/openshift/api/image/v1"
+
 	"github.com/openshift/openshift-apiserver/pkg/image/apis/image"
-	"github.com/openshift/openshift-apiserver/pkg/image/apis/image/docker10"
-	"github.com/openshift/openshift-apiserver/pkg/image/apis/image/dockerpre012"
+	imagedocker10helpers "github.com/openshift/openshift-apiserver/pkg/image/apis/image/docker10"
+	imagedockerpre012helpers "github.com/openshift/openshift-apiserver/pkg/image/apis/image/dockerpre012"
```
Comment (sttts) [inline pkg/image/apis/image/v1/register.go:13]: helpers is a strange package alias. Helpers with types?

File: pkg/apps/apis/apps/v1/conversion.go [hunk 7]
```diff
@@ -55,43 +55,44 @@ func Convert_v1_RollingDeploymentStrategyParams_To_apps_RollingDeploymentStrateg
 
 	if in.Pre != nil {
 		out.Pre = &newer.LifecycleHook{}
-		if err := Convert_v1_LifecycleHook_To_apps_LifecycleHook(in.Pre, out.Pre, s); err != nil {
+		if err := Convert_v1_LifecycleHook_To_apps_LifecycleHook(in.Pre, out.Pre, nil); err != nil {
```
Comment (sttts) [inline pkg/apps/apis/apps/v1/conversion.go:58]: why this?

File: pkg/build/apis/build/v1/conversion.go [hunk 8]
```diff
@@ -180,6 +181,90 @@ func Convert_v1_BinaryBuildRequestOptions_To_url_Values(in *v1.BinaryBuildReques
...
+func Convert_url_Values_To_v1_BuildLogOptions(in *url.Values, out *v1.BuildLogOptions, s conversion.Scope) error {
+	if in == nil || out == nil {
+		return nil
+	}
+	if values, ok := map[string][]string(*in)["container"]; ok && len(values) > 0 {
+		if err := runtime.Convert_Slice_string_To_string(&values, &out.Container, s); err != nil {
+			return err
+		}
+	} else {
+		out.Container = ""
```
Comment (sttts) [inline pkg/build/apis/build/v1/conversion.go:193]: how about `*out = v1.BuildLogOptions{}`? That would assign all the zero values.

File: pkg/build/apis/build/v1/conversion.go [hunk 9]
```diff
@@ -193,6 +278,11 @@ func AddCustomConversionFuncs(scheme *runtime.Scheme) error {
 	}); err != nil {
 		return err
 	}
+	if err := scheme.AddConversionFunc((*url.Values)(nil), (*v1.BuildLogOptions)(nil), func(a, b interface{}, scope conversion.Scope) error {
+		return Convert_url_Values_To_v1_BuildLogOptions(a.(*url.Values), b.(*v1.BuildLogOptions), scope)
+	}); err != nil {
+		return err
+	}
```
Comment (sttts) [inline pkg/build/apis/build/v1/conversion.go:285]: just return err directly

File: pkg/image/apis/image/v1/register.go (hunk 6 above)
Comment (dmage) [reply pkg/image/apis/image/v1/register.go:13]: yes, types live in openshift/api repo, and this repo has some helpers

File: pkg/image/apis/image/docker10/types.go (hunk 5 above)
Comment (dmage) [reply pkg/image/apis/image/docker10/types.go:9]: yes, about 2 years ago
https://github.com/openshift/api/blob/master/image/docker10/dockertypes.go#L11

File: pkg/apps/apis/apps/v1/conversion.go (hunk 7 above)
Comment (p0lyn0mial) [reply pkg/apps/apis/apps/v1/conversion.go:58]: I will revert.

File: pkg/apps/apis/apps/v1/conversion.go [hunk 10]
```diff
@@ -55,43 +55,44 @@ func Convert_v1_RollingDeploymentStrategyParams_To_apps_RollingDeploymentStrateg
...
 			return err
 		}
 	}
 	if in.Post != nil {
 		out.Post = &newer.LifecycleHook{}
-		if err := Convert_v1_LifecycleHook_To_apps_LifecycleHook(in.Post, out.Post, s); err != nil {
+		if err := Convert_v1_LifecycleHook_To_apps_LifecycleHook(in.Post, out.Post, nil); err != nil {
```
Comment (p0lyn0mial) [inline pkg/apps/apis/apps/v1/conversion.go:64]: @sttts what about others in this pkg? I think I'm going to revert them as well.

File: pkg/build/apis/build/v1/conversion.go (hunk 8 above)
Comment (p0lyn0mial) [reply pkg/build/apis/build/v1/conversion.go:193]: yes, that will work.

**Pull request merged**